File: src/main/java/cz/vhromada/catalog/web/mapper/EpisodeMapper.java
package cz.vhromada.catalog.web.mapper;

import cz.vhromada.catalog.entity.Episode;
import cz.vhromada.catalog.web.fo.EpisodeFO;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for episodes.
 *
 * @author Vladimir Hromada
 */
@Mapper(uses = TimeMapper.class)
public interface EpisodeMapper {

    /**
     * Returns FO for episode.
     *
     * @param source episode
     * @return FO for episode
     */
    EpisodeFO map(Episode source);

    /**
     * Returns episode.
     *
     * @param source FO for episode
     * @return episode
     */
    Episode mapBack(EpisodeFO source);

}

File: src/main/kotlin/cz/vhromada/catalog/web/mapper/EpisodeMapper.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Episode
import cz.vhromada.catalog.web.fo.EpisodeFO

/**
 * An interface represents mapper for episodes.
 *
 * @author Vladimir Hromada
 */
interface EpisodeMapper {

    /**
     * Returns FO for episode.
     *
     * @param source episode
     * @return FO for episode
     */
    fun map(source: Episode): EpisodeFO

    /**
     * Returns episode.
     *
     * @param source FO for episode
     * @return episode
     */
    fun mapBack(source: EpisodeFO): Episode

}
